pyuvdata/mwa_corr_fits.py

```
@@ -0,0 +1,482 @@
+                        # unflag where the data is
+                        self.flag_array[:, bls_ind, :, pol_ind] = False
+                        # nsamples = 1 where the data is
+                        self.nsample_array[:, bls_ind, :, pol_ind] = 1.0
+
+        # generage baseline flags for flagged ants
+        bad_ant_inds = []
+        for ant1 in range(128):
+            for ant2 in range(ant1, 128):
+                if ant1 in flagged_ants or ant2 in flagged_ants:
+                    bad_ant_inds.append(int(128 * ant1 - ant1 * (ant1 + 1) / 2 + ant2))
+
+        self.flag_array[:, bad_ant_inds, :, :] = True
+
+        # combine baseline and time axes
+        self.data_array = self.data_array.reshape((self.Nblts, self.Nfreqs, self.Npols))
+        self.flag_array = self.flag_array.reshape((self.Nblts, self.Nfreqs, self.Npols))
+        self.nsample_array = self.nsample_array.reshape((self.Nblts, self.Nfreqs, self.Npols))
+
+        # cable delay corrections
+        if correct_cable_len is True:
+            self.correct_cable_length(cable_lens)
+
+        # add spectral window index
+        self.data_array = self.data_array[:, np.newaxis, :, :]
+        self.flag_array = self.flag_array[:, np.newaxis, :, :]
+        self.nsample_array = self.nsample_array[:, np.newaxis, :, :]
+
+        # because of an annoying discrepancy between file conventions, in order
+        # to be consistent with the uvw vector direction, all the data must
+        # be conjugated
+        self.data_array = np.conj(self.data_array)
+
+        # reorder polarizations
+        self.reorder_pols()
+
+        # phasing
+        if phase_data is True:
+            (ra, dec) = pointing_center
+            self.phase(ra, dec)
+
+        # TODO: add support for cotter flag files
+        if use_cotter_flags is True:
+            warnings.warn('reading in cotter flag files is not yet available')
```

I think this should raise a `NotImplementedError` rather than a warning.